src/main/java/duke/Command.java

```diff
@@ -0,0 +1,252 @@
+            taskList.addTask(t);
+            ui.printTaskAdded(t, taskList.getCount());
+            try {
+                storage.appendTaskToFile(t);
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+
+        case "event":
+            temp = args.split(" /at ", 2);
+            t = new Event(temp[0], LocalDate.parse(temp[1],
+                    DateTimeFormatter.ofPattern(INPUT_DATE_FORMAT)));
+            taskList.addTask(t);
+            ui.printTaskAdded(t, taskList.getCount());
+            try {
+                storage.appendTaskToFile(t);
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+
+        case "todo":
+            t = new ToDo(args);
+            taskList.addTask(t);
+            ui.printTaskAdded(t, taskList.getCount());
+            try {
+                storage.appendTaskToFile(t);
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+
+        case "mark":
+
+        case "unmark":
+            index = Integer.parseInt(args) - 1;
+            t = taskList.getTasks().get(index);
+            boolean b = name.equals("mark");
+            taskList.markTask(t, b);
+            ui.printTaskMarked(t, b);
+            try {
+                storage.rewriteFile(taskList.getTasks());
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+        }
+    }
+
+    private boolean isValidDate(String date) {
+        try {
+            LocalDate d = LocalDate.parse(date, DateTimeFormatter.ofPattern(INPUT_DATE_FORMAT));
+            return true;
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+    }
+
+    private boolean isCorrectUsage(String args, Duke duke) {
+        UI ui = duke.getUI();
+        TaskList taskList = duke.getTaskList();
+        String usage = getCorrectUsage();
+        switch(name) {
+        case "delete":
+            if (args.isEmpty()) {
+                ui.printError("Please specify a task number.\n\n" + usage);
+                return false;
+            }
+            return true;
+
+        case "find":
+            if (args.isBlank()) {
+                ui.printError("Please specify at least one keyword.\n\n" + usage);
+                return false;
+            } else if (args.charAt(0) != '"' || args.charAt(args.length() - 1) != '"') {
+                if (args.split(" ").length > 1) {
+                    ui.printError("'find' only expects 1 argument.\n\n If there are multiple "
+                            + "keywords, please enclose them in quotation marks (\"\").");
+                    return false;
+                }
+            } else {
+                if (args.length() == 2 || args.substring(1, args.length() - 1).isBlank()) {
```

Comment:
This condition is a bit complex, it might help to abstract some of this information into a method or evaluate these statements in separate intemediate steps?